quickjs.c

```diff
@@ -3980,6 +3994,18 @@ JSValue JS_NewAtomString(JSContext *ctx, const char *str)
     return val;
 }
 
+uint8_t *JS_ToCStringLenRaw(JSContext *ctx, size_t *plen, JSValueConst val)
+{
+    if (JS_VALUE_GET_TAG(val) != JS_TAG_STRING)
+    {
+        if(plen) *plen = 0;
+        return NULL;
+    }
+    JSString *str = JS_VALUE_GET_STRING(val);
+    if(plen) *plen = str->len;
+    return str->u.str8;
+}
+
```

Comment:
This one has a problem: the caller cannot determine if the string was encoded as 8-bit or 16-bit.  Furthermore the name is misleading: the returned pointer is not UTF-8 encoded if the string has non-ASCII characters.